### PR TITLE
14 create new detector

### PR DESCRIPTION
**Pull request by ao-picterra**

### CODE REVIEW

File: examples/training.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import json
+from picterra import APIClient
+
+# Put your API key either:
+# * replacing '1234'
+# * not putting arguments to APIClient but setting the PICTERRA_API_KEY environment variable 
+client = APIClient(api_key='59c4715cee86d854e233739260fdfd8373f13fc393ee66ca389b04bc40e80472', base_url='https://app-testing.picterra.ch/public/api/v1/')
```
Comment (julienr) [inline examples/training.py:7]: I think we should remove the api_key here :-)

File: src/picterra/client.py [hunk 2]
```diff
@@ -307,3 +312,13 @@ def set_annotations(self, detector_id, raster_id, annotation_type, annotations):
...
+
+    def train_detector(self, detector_id):
+        """
+        Start the training of a detector
+
+        Args:
+            detector_id (str): The id of the detector
+        """
+        resp = self.sess.post(self._api_url('detectors/%s/train/' % detector_id))
+        assert resp.status_code == 201, resp.status_code
```
Comment (julienr) [inline src/picterra/client.py:324]: this should poll for the operation to be finished

File: tests/test_client.py [hunk 3]
```diff
@@ -294,5 +300,8 @@ def test_upload_annotations():
 @responses.activate
 def test_train_detector():
     add_mock_detector_train_responses(1)
+    add_mock_operations_responses('pending')
```
Comment (julienr) [inline tests/test_client.py:303]: IIRC, we don't expose pending in the public API: we only have running, success and failed

**Review by julienr — CHANGES_REQUESTED** [review]
LGTM otherwise

File: src/picterra/client.py [hunk 4]
```diff
@@ -273,9 +273,14 @@ def set_annotations(self, detector_id, raster_id, annotation_type, annotations):
         Args:
             detector_id (str): The id of the detector
             raster_id (str): The id of the raster
-            annotation_type (str): One of (outlines, training_area, testing_area, validation_area)
+            annotation_type (str): One of (outline, training_area, testing_area, validation_area)
```
Comment (restaste) [inline src/picterra/client.py:276]: Ouch, good catch!

File: tests/test_client.py [hunk 5]
```diff
@@ -37,8 +39,20 @@ def add_mock_detector_creation_response():
     responses.add(responses.POST, api_url('detectors/'), json={'id': 'foobar'}, status=201)
 
 
+def add_mock_detector_train_responses(detector_id):
+    operation_id = OPERATION_ID
```
Comment (restaste) [inline tests/test_client.py:43]: nit: I guess you can inline that field, to follow the same usage as `TEST_POLL_INTERVAL`

File: tests/test_client.py (hunk 3 above)
Comment (restaste) [reply tests/test_client.py:303]: The current `/operations/` endpoint returns `job.status` (so `pending` can be returned). I agree that might have no value to the user (plus, if they see it `pending`, users might wrongly believe an action is needed on their part). I'll fix that in the public api.

Comment (restaste) [reply tests/test_client.py:303]: https://gitlab.com/picterra/platform/-/merge_requests/2241

Comment (julienr) [reply tests/test_client.py:303]: Ah I thought we already did that. Thanks !

**Review by julienr — APPROVED** [review]

**Pull request merged**